scheduler/models/scheduled_job.py

```diff
@@ -156,16 +156,27 @@ def ready_for_schedule(self) -> bool:
             return False
         return True
 
-    def schedule(self) -> bool:
-        if not self.ready_for_schedule():
+    def schedule(self, when=None) -> bool:
+        if not self.ready_for_schedule() and when is None:
             return False
-        schedule_time = self._schedule_time()
+        schedule_time = self._schedule_time() if when is None else when
```

Comment:
I don't understand. I made the schedule() to accept either no params or a datetime. Because doing it in a separate function would mean repeating code. I can do it if needed.

So, in my PR, following with the idea I proposed, I renamed `run_job_now` to `enqueue_job_now`. And created a new `run_jub_now`.

The rationale behind is: you don't want jobs that are scheduled in the next five minutes to be run "now" and then later in 5 minutes (such as sending an email).
I was facing more the opposite issue, a job scheduled to run in 6 hours, and choosing "run now" would do nothing.

Probably a better name for `enqueue_job_now` would be `schedule_job_now`. I'm happy to rename it if you want!

Relevant code:
<img width="782" alt="image" src="https://user-images.githubusercontent.com/96071233/226675401-29a28a30-0a05-48e7-bdd4-d2a4cb274804.png">

Reply:
renamed `enqueue_job_now` -> `schedule_job_now `

Reply:
Sorry, I am still not following why have `when` param? isn't sending with `when=now` the same as calling `enqueue_to_run`?

Reply:
Well, no, and that exact thing is what prompted me to make this PR in the first place.
when you straight enqueue (or enqueue_to_run), you are putting the job in the queue, period. the worker will do the rest.
when you put `when=now` to the `schedule` method, it will add it to the scheduled registry, and that would then require the scheduler task to move it to the queue. That would be acceptable, if not because the schedule method has a de-duplication logic that would not allow a scheduled job to be scheduled twice. So, if the job is scheduled to run in 2 hours, it will `not` heed the order to `schedule(when=now)`. You can try it...

Reply:
(unless this was modified from 2023.2.1)

Reply:
Why is this needed then? please remove it.

Reply:
I'd be glad to put a hold on this until I make a few more tests, since it seems that current "Run Now" works (It is not the case for 2023.1.x>= V <= 2023.2.x ).

Reply:
Ok, now the only change that remains is using the new way of running jobs (and schedule can schedule with an optional timestamp). Direct Enqueuing was removed.
Let me know!

Reply:
Can we remove the `when` parameter from here so we can merge this?

Reply:
I did. you can check again.
Worth mentioning that I had to add a new method called `schedule_now`. Hope this can be merged soon.
Thanks!